### PR TITLE
Declare Python 3.13 compatibility

### LINKED ISSUE #1176: Python 3.13 support release blockers
Before declaring official support for Python 3.13 the following issues have to be resolved and releases published afterwards:

- [x] https://github.com/zopefoundation/RestrictedPython/issues/264
- [x] https://github.com/zopefoundation/AccessControl/pull/156
- [x] https://github.com/zopefoundation/zope.interface/pull/322 - strange test failures and incompatibilites between local and GHA runs
- [x] https://github.com/zopefoundation/Zope/pull/1215 – collect all the new releases
- [x] https://github.com/zopefoundation/zope.testrunner/issues/155
- [x] https://github.com/zopefoundation/persistent/pull/188 – requires a `cffi` release
- [x] https://github.com/zopefoundation/zope.interface/issues/279 – currently also blocked by `cffi`
- [x] https://github.com/zopefoundation/Acquisition/pull/68 – test failures
- [x] https://github.com/zopefoundation/AccessControl/pull/143 – cannot be build without Acquisition fixed first
- [x] https://github.com/zopefoundation/zodbpickle/pull/81 – segfault in tests
- [x] https://github.com/zopefoundation/zope.security/pull/104 – segfault in tests
- [x] https://github.com/zopefoundation/zope.security/pull/105 Wheel does not build on MacOS
- [x] https://github.com/zopefoundation/BTrees/pull/202
- [x] https://github.com/zopefoundation/AccessControl/pull/154
- [x] https://github.com/zopefoundation/zope.security/pull/110
- [x] https://github.com/zopefoundation/zodbpickle/pull/83 Compilation errors in C code
- [x] https://github.com/zopefoundation/zope.container/pull/53
- [x] https://github.com/zopefoundation/ZConfig/pull/97 test failure
- [x] ~~https://github.com/zopefoundation/zope.testrunner/issues/167 test failures~~ The fact that the zope.testrunner tests themselves fail on Python 3.13 is not a release blocker for Zope's Python 3.13 support if Zope's own tests run fine on Python 3.13.

### PR DESCRIPTION
**Pull request by dataflake**
Fixes #1176 

This PR declares Python 3.13 support and updates dependencies to Python 3.13-compatible versions.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -17,7 +17,7 @@ envlist =
 [testenv]
 skip_install = true
 deps =
-    setuptools < 74
+    setuptools  <74
```
Comment (d-maurer) [inline tox.ini:20]: Should we keep the blank?

File: tox.ini [hunk 2]
```diff
@@ -50,12 +48,36 @@ commands =
...
+    universal2: MarkupSafe
+    universal2: zope.testrunner
+
 
 [testenv:release-check]
 description = ensure that the distribution is ready to release
 basepython = python3
 skip_install = true
 deps =
+    setuptools <74
```
Comment (d-maurer) [inline tox.ini:80]: Usually, we have blanks around comparison operators

**Review by d-maurer — APPROVED** [review]

File: tox.ini (hunk 2 above)
Comment (dataflake) [reply tox.ini:80]: "Whitespace between a conditional operator and the following version identifier is optional, as is the whitespace around the commas.", see https://packaging.python.org/en/latest/specifications/version-specifiers/#id5

Comment (dataflake) [reply tox.ini:80]: The definition for that specification is in zope.meta, not in this package.

File: tox.ini (hunk 1 above)
Comment (dataflake) [reply tox.ini]: I've cleaned up the spacing in zope.meta and applied it here.

File: tox.ini (hunk 2 above)
Comment (dataflake) [reply tox.ini]: I have cleaned up the spacing in zope.meta and applied it here.

File: CHANGES.rst [hunk 3]
```diff
@@ -10,6 +10,10 @@ https://github.com/zopefoundation/Zope/blob/4.x/CHANGES.rst
 5.11 (unreleased)
 -----------------
 
+- Upgrade all dependencies to versions supporting Python 3.13.
+
+- Add support for Python 3.13.
+
 - Drop support for Python 3.7.
 
 - Update to newest compatible versions of dependencies.
```
Comment (icemac) [inline CHANGES.rst:19]: This line now seems to duplicate the first one.

**Review by icemac — APPROVED** [review]

File: CHANGES.rst (hunk 3 above)
Comment (dataflake) [reply CHANGES.rst:17]: I removed it.

**Comment by mauritsvanrees** [comment]
I ran the Plone 6.1 tests with Zope master and its versions and all was fine.
Well, one test failure on Python 3.10 probably due to an updated dependency, and we need some newer version pins before Python 3.13 can work, but that is something to fix on the Plone side.

I was going to say: "from my POV Zope is ready for release" but I see you just did that. :-D
Welcome Zope 5.11!

Thanks all!

**Pull request merged**